.pre-commit-config.yaml

```diff
@@ -603,6 +603,14 @@ repos:
         language: python
         files: ^.*LICENSE.*$|^.*LICENCE.*$
         pass_filenames: false
+      - id: check-aiobotocore-optional
+        name: Check if aiobotocore is an optional dependency only
+        entry: ./scripts/ci/pre_commit/pre_commit_check_aiobotocore_optional.py
+        language: python
+        files: ^airflow/providers/.*/provider\.yaml$
```

Comment:
It should run only on the amazon provider yaml does it not?

Reply:
I had the same thought but then wondered that we have some transfer operators too like `s3_to_gcs` in the google providers for example and felt we might need this check there too 🤔

Reply:
That was exactly my point - it costs us almost nothing to exclude aiobotocore from all providers.